modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -165,9 +165,7 @@ export default class TileLayer<DataT = any, ExtraPropsT extends {} = {}> extends
   }
 
   get isLoaded(): boolean {
-    return this.state?.tileset?.selectedTiles?.every(
-      tile => tile.isLoaded && tile.layers && tile.layers.every(layer => layer.isLoaded)
```

Comment:
An MVTLayer test is failing in the picking module when layer cache checking is pushed into Tileset2D. 

For background, the TileLayer [caches sub layers](https://github.com/visgl/deck.gl/pull/8290/files#diff-8624e1d990265f2e01a5dce4cf739482733edb375fdd8a0df5c5c7df8c30f16fL342) rendered by each tile in the Tileset2D `tiles[...].tile.layers` property. 

Tileset2D#isLoaded exists but only checks if tile data is loaded, whereas TileLayer#isLoaded checks if these layers as well as if the tile data is loaded. Oddly, TileLayer doesn't use Tileset2D#isLoaded. This seemed like an odd choice for TileLayer - why doesn't TileLayer just use Tileset2D's isLoaded?

There's an edge case too. TileLayer is "loaded" if all layers rendered on a tile are present and loaded. I'm not sure what the design intention is for `tile.layers`. It starts null, then eventually it's defined and contains layers.

Alternatively, can we remove the Tileset2D#isLoaded method and just rely on the layer to implement it? I can't find usages outside of tests.

Reply:
I'm moving these changes into #8335 since they're blocking other fixes.